src/helpers.coffee

```diff
@@ -179,7 +188,44 @@ exports.keydownEvent = (key, {ctrl, shift, alt, cmd, keyCode, target, location}=
   Object.defineProperty(event, 'which', get: -> keyCode)
   event
 
+# bindingKeystrokes and userKeystrokes are arrays of keystrokes
+# e.g. ['crtl-y', 'ctrl-x', '^x']
+exports.keystrokesMatch = (bindingKeystrokes, userKeystrokes) ->
+  userKeystrokeIndex = -1
+  userKeystrokesHasKeydownEvent = false
+  matchesNextUserKeystroke = (bindingKeystroke) ->
+    while userKeystrokeIndex < userKeystrokes.length - 1
+      userKeystrokeIndex += 1
+      userKeystroke = userKeystrokes[userKeystrokeIndex]
+      isKeydownEvent = not userKeystroke.startsWith('^')
+      userKeystrokesHasKeydownEvent = true if isKeydownEvent
+      if bindingKeystroke is userKeystroke
+        return true
+      else if isKeydownEvent
+        return false
+    null
+
+  for bindingKeystroke in bindingKeystrokes
+    doesMatch = matchesNextUserKeystroke(bindingKeystroke)
+    if doesMatch is false
+      return false
+    else if doesMatch is null
+      # Make sure userKeystrokes with only keyup events doesn't match everything
```

Comment:
That makes sense. This happens when a keydown matches a binding, then the queue is cleared. When the keyup comes in, it will be the only thing in the queue, and :boom:. 

I kind of like the check in here, though (maybe in addition to another check somewhere else?), because it makes the function feel logically complete.